lib/reek/tree_walker.rb

```diff
@@ -18,24 +18,35 @@ module Reek
   #
   # :reek:TooManyMethods: { max_methods: 29 }
   class TreeWalker
-    def initialize(smell_repository, exp)
-      @smell_repository = smell_repository
-      @exp = exp
+    private_attr_accessor :element
+    private_attr_reader :exp, :context_tree
+
+    def initialize(syntax_tree)
+      @exp = syntax_tree
       @element = Context::RootContext.new(exp)
       @context_tree = process(exp)
     end
 
-    def walk
-      context_tree.each do |element|
-        smell_repository.examine(element)
+    # Walks the syntax tree, runs the configured SmellDetectors and returns
+    # a SmellRepository that contains the reported SmellWarnings.
+    #
+    # @param smell_types [Array] The smell types we want to apply when walking the tree.
+    # @param configuration [Configuration::AppConfiguration] The corresponding configuration.
+
+    #
+    # @return [Reek::Smells::SmellRepository] SmellRepository that contains all used
+    #         SmellDetectors including all reported SmellWarnings.
+    def walk(smell_types: raise, configuration: raise)
```

Comment:
Yes, I did change the signature of "walk" as well. I always hated those parameter-less methods, feels like 1965.
I believe this change is a step forward since it allows us for more flexible uses of the TreeWalker in the future.
E.g. Walking the same tree but using different configurations and then comparing the results and doing other awesome stuff like this ;)
And it just feels more fluent to me.